kernel/printk.c

```diff
@@ -16,10 +16,24 @@ static void co_vsnprintf(const char *format, va_list ap)
     retval = vsnprintf(vsnprintf_buf, VSNPRINTF_BUF_SIZE, format, ap);
 }
 
-void __printk_putchar(char c);
-
+#include <serial.h>
 #include "platform.h"
 
+void __printk_init(void)
+{
+    usart_init();
```

Comment:
should be `uart_init`.

Reply:
I found [`usart_init`](https://elixir.free-electrons.com/linux/latest/source/drivers/tty/serial/stm32-usart.c#L1020) in Linux v4.13.7. But It's only for stm32 family Indeed. In my opinion, I would follow the generic way use `uart_init` instead of `usart_init`. Any suggestion?

Reply:
I agree with the consistent `uart` rather than `usart`.